src/requestforattestation/RequestForAttestation.spec.ts

```diff
@@ -115,6 +124,54 @@ describe('RequestForAttestation', () => {
     expect(request.claim.owner).toBeUndefined()
   })
 
+  it('compresses and decompresses the request for attestation object', () => {
+    const legitimationAttestationBob: Attestation = Attestation.fromRequestAndPublicIdentity(
+      legitimationRequest,
+      identityBob
+    )
+    const legitimationBob: AttestedClaim = AttestedClaim.fromRequestAndAttestation(
+      legitimationRequest,
+      legitimationAttestationBob
+    )
+    const reqForAtt: RequestForAttestation = buildRequestForAttestation(
+      identityBob,
+      'ctype',
+      {
+        a: 'a',
+        b: 'b',
+        c: 'c',
+      },
+      [legitimationCharlie, legitimationBob]
+    )
+
+    const sortedCompressedReqForAtt: CompressedRequestForAttestation = [
+      compressClaim(reqForAtt.claim),
+      compressClaimHashTree(reqForAtt.claimHashTree),
+      compressNonceAndHash(reqForAtt.claimOwner),
+      reqForAtt.claimerSignature,
+      compressNonceAndHash(reqForAtt.cTypeHash),
+      reqForAtt.rootHash,
+      compressLegitimation(reqForAtt.legitimations),
+      reqForAtt.delegationId,
+    ]
+
+    expect(compressRequestForAttestation(reqForAtt)).toEqual(
+      sortedCompressedReqForAtt
+    )
+
+    expect(decompressRequestForAttestation(sortedCompressedReqForAtt)).toEqual(
+      reqForAtt
+    )
+
+    const decompressedRequestForAttestationObj = decompressRequestForAttestation(
+      sortedCompressedReqForAtt
+    )
+
+    expect(
+      RequestForAttestation.decompress(decompressedRequestForAttestationObj)
+    ).toEqual(reqForAtt)
+  })
```

Comment:
add negative tests

Reply:
updated.

Reply:
Please rethink the negative tests

Reply:
I think this is better. I have added checks. overall